torch/_dynamo/device_interface.py

```diff
@@ -171,6 +171,73 @@ def get_compute_capability(device: _device_t = None):
         return major * 10 + min
 
 
+get_xpu_stream: Optional[Callable[[int], int]]
+if torch.xpu._is_compiled():
+    from torch._C import _xpu_getCurrentRawStream as get_xpu_stream
+else:
+    get_xpu_stream = None
+
+
+class XPUInterface(DeviceInterface):
+    device = torch.xpu.device
+    Event = torch.xpu.Event
+    Stream = torch.xpu.Stream
+
+    class Worker:
+        @staticmethod
+        def set_device(device: int):
+            caching_worker_current_devices["xpu"] = device
+
+        @staticmethod
+        def current_device() -> int:
+            if "xpu" in caching_worker_current_devices:
+                return caching_worker_current_devices["xpu"]
+            return torch.xpu.current_device()
+
+        @staticmethod
+        def get_device_properties(device: _device_t = None):
+            if device is not None:
+                if isinstance(device, str):
+                    device = torch.device(device)
+                    assert device.type == "xpu"
+                if isinstance(device, torch.device):
+                    device = device.index
+            if device is None:
+                device = XPUInterface.Worker.current_device()
+
+            if "xpu" not in caching_worker_device_properties:
+                device_prop = [
+                    torch.xpu.get_device_properties(i)
```

Comment:
@guangyey , currently, the python interface does not expose some properties that Triton depends on. May I know if the `torch._C` can get these properties? If not, could you help submit a PR to add these properties?

Reply:
We need a PR to fetch these properties.

Reply:
@guangyey , thanks for the information. Could you please submit a PR ASAP? With your PR being landed, this PR is just read for review.

Reply:
OKey